src/ng/http.js

```diff
@@ -1293,7 +1297,7 @@ function $HttpProvider() {
           reqHeaders[(config.xsrfHeaderName || defaults.xsrfHeaderName)] = xsrfValue;
         }
 
-        $httpBackend(config.method, url, reqData, done, reqHeaders, config.timeout,
+        $httpBackend(config.method, (url === $sce.valueOf(origUrl) ? origUrl : url), reqData, done, reqHeaders, config.timeout,
```

Comment:
So this line is weird since we it is saying: "if the newly built url is not the same as the original one then just use the original one. But this defeats the purpose of being able to provide parameters, etc.
But if we pass the untrusted URL through it should not be allowed by SCE inside the httpBackend, which now expects trusted URLs for JSONP.
And if we "retrust" the built URL for JSONP requests we are undermining the security unless we can ensure that the `buildUrl` function cannot modify the URL in a way that allows an attack.

Given all of this, it might be easier to say that `$http` can only take strings as URLs, which is what is suggested by @klhuillier in https://github.com/angular/angular.js/issues/11328#issuecomment-81778364. That way the only way to get JSONP requests to work is if the request URL is in the SCE resource URL whitelist.